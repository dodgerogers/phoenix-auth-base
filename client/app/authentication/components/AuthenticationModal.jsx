import React from 'react'
import PropTypes from 'prop-types';
import { Button, Divider, Modal, Message } from 'semantic-ui-react';
import { NotificationContainer, areaIDs } from '../../common/Notifications';


const AuthenticationModal = (props) => {
  return (
    <Modal
      size="mini"
      dimmer={false}
      open={props.show}
      closeOnDocumentClick={true}
      onClose={props.close}
      >
      <Modal.Header as="h2">
        {props.title}
      </Modal.Header>
      <Modal.Content>
        <NotificationContainer id={areaIDs.AUTHENTICATION} />
        {props.children}
      </Modal.Content>
      <Modal.Actions>
        <Button onClick={props.close}>
          Close
        </Button>
      </Modal.Actions>
    </Modal>
  );
}

AuthenticationModal.propTypes = {
  show: PropTypes.bool,
  close: PropTypes.func.isRequired,
  title: PropTypes.string,
}

export default AuthenticationModal;
