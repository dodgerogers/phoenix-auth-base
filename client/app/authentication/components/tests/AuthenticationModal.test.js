import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import AuthenticationModal from '../AuthenticationModal';


describe('AuthenticationModal', () => {
  const fullProps = {
    close: jest.fn(),
  };

  const pureComponent = (props) => shallow(
    <AuthenticationModal {...props} title="Auth modal" />
  );

  it('matches snapshot', () => {
    const wrapper = pureComponent(fullProps);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
