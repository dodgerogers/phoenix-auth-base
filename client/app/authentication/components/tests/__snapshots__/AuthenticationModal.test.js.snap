// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthenticationModal matches snapshot 1`] = `
<Modal
  centered={true}
  closeOnDimmerClick={true}
  closeOnDocumentClick={false}
  dimmer={true}
  eventPool="Modal"
  onClose={[Function]}
  size="mini"
>
  <ModalHeader
    as="h2"
  >
    Auth modal
  </ModalHeader>
  <ModalContent>
    <Connect(NotificationList)
      id="AUTHENTICATION"
    />
  </ModalContent>
  <ModalActions>
    <Button
      as="button"
      onClick={[Function]}
    >
      Close
    </Button>
  </ModalActions>
</Modal>
`;
